core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -873,6 +875,36 @@ object Chunk {
       case _: BitChunk       => ClassTag.Boolean.asInstanceOf[ClassTag[A]]
     }
 
+  private val bufferSize: Int =
+    100
+
+  private final case class Add[A](start: Chunk[A], buffer: Array[AnyRef], bufferUsed: Int, chain: AtomicInteger)
+      extends Chunk[A] {
+
+    implicit val classTag: ClassTag[A] = classTagOf(start)
+
+    val length: Int =
+      start.length + bufferUsed
+
+    override def +[A1 >: A](a1: A1): Chunk[A1] =
+      if (bufferUsed < buffer.length && chain.compareAndSet(bufferUsed, bufferUsed + 1)) {
+        buffer(bufferUsed) = a1.asInstanceOf[AnyRef]
+        Add(start, buffer, bufferUsed + 1, chain)
+      } else {
+        val buffer = Array.ofDim[AnyRef](bufferSize)
+        buffer(0) = a1.asInstanceOf[AnyRef]
+        Add(start, buffer, bufferUsed, chain)
+      }
+
+    def apply(n: Int): A =
+      if (n < start.length) start(n) else buffer(n - start.length).asInstanceOf[A]
+
+    protected[zio] def toArray[A1 >: A](n: Int, dest: Array[A1]): Unit = {
+      start.toArray(n, dest)
+      val _ = buffer.asInstanceOf[Array[A]].copyToArray(dest, n + start.length, bufferUsed)
```

Comment:
Doesn't `copyToArray` return `Unit`?

Reply:
On Scala 2.13 it returns an integer with the number of elements written to the array.

Reply:
Ah, I see, my bad.